river/Output.zig

```diff
@@ -56,19 +67,23 @@ usable_box: Box,
 views: ViewStack(View) = .{},
 
 /// The double-buffered state of the output.
-current: State = State{ .tags = 1 << 0 },
-pending: State = State{ .tags = 1 << 0 },
+current: State = State{ .tags = 1 << 0, .layout = false },
+pending: State = State{ .tags = 1 << 0, .layout = false },
 
 /// Number of views in "main" section of the screen.
 main_count: u32 = 1,
 
 /// Percentage of the total screen that the "main" section takes up.
 main_factor: f64 = 0.6,
 
-/// Current layout of the output. If it is "full", river will use the full
-/// layout. Otherwise river assumes it contains a string which, when executed
-/// with sh, will result in a layout.
-layout: []const u8,
+/// Current layout-namespace of the output.
+layout_namespace: ?[]const u8 = null,
+
+/// The currently active LayoutDemand
+layout_demand: ?*LayoutManager.LayoutDemand = null,
```

Comment:
Is there a reason this needs to be a pointer? It seems to me that we could get rid of the indirection and allocation here:
```suggestion
layout_demand: ?LayoutManager.LayoutDemand = null,
```